### DISCUSSION TITLE
Spatially dependent coefficient of thermal expansion (CTE) for calculating eigen strain

### DISCUSSION THREAD

**Discussion by chakra34** [idaholab/moose#24417] [Q&A General]
Following the discussion in , https://github.com/idaholab/moose/discussions/21785, which considers temperature dependent CTE, I was wondering whether I can also pass this material property as a function of space, for example linearly varying with coordinate 'z'.

To do this I tried the following:
```
[Materials]
  [./elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 150e6
    poissons_ratio = 0.3
  [../]
  [./small_stress]
    type = ComputeFiniteStrainElasticStress
  [../]
  
  [./thermal_expansion_strain]
    type = ComputeInstantaneousThermalExpansionFunctionEigenstrain
    thermal_expansion_function = cte_func_inst
    stress_free_temperature = 0.0
    temperature = temp
    eigenstrain_name = eigenstrain
  [../]
[]

[Functions]

  [./cte_func_inst]
    type = ParsedFunction
    value = '1e-4*(z+1)'
  [../]

  [./temp_function]
    type  = ParsedFunction
    value = 100.0
  [../]
 
[]
```
But I don't see any difference in the eigenstrain, most likely because the CTE should be a function of temperature and not space. I was wondering whether it is possible to allow the CTE to be position dependent via using ParsedFunc/ParsedAux/ParsedMaterial.

I did try using ParsedMaterial and write the above expression, but it also gave an error, which in my understanding also points to the notion that the ParsedMaterial function should also be dependent on a coupled variable (temperature) instead of space.

In my actual problem I will have a time and space varying temperature field in the body as well as was hoping for a spatially variable CTE.

Please let me know your thoughts, and again any help is greatly appreciated :) 

-Aritra

**Comment by GiudGiud** [comment]
I dont think this is can be done through the input file. 
The creators of this object meant to model the influence of temperature. The functions are queried with a temperature argument, not a spatial coordinate.

You can easily adapt the source code to use Z or any function of Z though. So in that way yes it is very possible to allow the CTE to depend on position.

Guillaume
